### PR TITLE
[NR-55782] Release chart with release toolkit

### PR DESCRIPTION
**Pull request by paologallinaharbur**
Merging this PR we will have a new process to release the chart:
 - changelog is updated automatically with release notes ([that should be part of the package](https://github.com/helm/chart-releaser/pull/217))
 - version is bumped automatically according to the changelog

As soon as well feel comfortable (even now if we want), we can automatize this easily by adding the `schedule` clause in the triggers.

The pipeline has been tested [here](https://github.com/paologallinaharbur/newrelic-prometheus-configurator/pull/1), few releases were created already in the fork from the changelog and dependabot.

My only concern with the approach is if it was easier so install `rt` and run the make, instead of having the logic twice being unable to test the code locally

### CODE REVIEW

**Review by kang-makes — COMMENTED** [review]
We will need to do a way harder pipeline to avoid having a circular dependency, properly test that we need a new version, and tag the commit that is really meant to be tagged.

But as the first iteration, it is to find these issues. I really like how it looks.

File: .github/workflows/releaseChart.yaml
Comment (paologallinaharbur) [reply .github/workflows/releaseChart.yaml:26]: my bad! thanks for noticing

File: .github/workflows/releaseChart.yaml
Comment (paologallinaharbur) [reply .github/workflows/releaseChart.yaml:9]: right now it should be an issue just if you click twice, but it is better to add it!

File: .github/workflows/releaseChart.yaml
Comment (paologallinaharbur) [reply .github/workflows/releaseChart.yaml:63]: nice, I did not know, I'll add it

File: .github/workflows/releaseChart.yaml
Comment (gsanchezgavier) [inline .github/workflows/releaseChart.yaml:3]: would make sense to create another separate target for the chart?

**Review by gsanchezgavier — COMMENTED** [review]
Looks good to me, I think we can give it a try. Could you write a short document like [this](https://github.com/newrelic/newrelic-prometheus-configurator/pull/126/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5) for anyone that wants to release the chart can follow ?

File: .github/workflows/releaseChart.yaml
Comment (gsanchezgavier) [inline .github/workflows/releaseChart.yaml:8]: would make sense to add some optional parameters that would skip the rt steps for any time we need to release without this tool?

File: .github/workflows/releaseChart.yaml
Comment (paologallinaharbur) [reply .github/workflows/releaseChart.yaml:3]: There is a separate target `make release-changelog-chart`. It was a copy-paste issue

File: .github/workflows/releaseChart.yaml
Comment (paologallinaharbur) [reply .github/workflows/releaseChart.yaml:84]: We discussed it online. Actually, the chart-releaser-action is creating the release using the latest commit as reference, therefore it should be not an issue.

File: .github/workflows/releaseChart.yaml
Comment (paologallinaharbur) [inline .github/workflows/releaseChart.yaml:6]: we are adding a check to be sure we are running over main, @gsanchezgavier is taking care of it

File: .github/workflows/releaseChart.yaml
Comment (paologallinaharbur) [reply .github/workflows/releaseChart.yaml:8]: discussed it offline, no for now

File: .github/workflows/releaseChart.yaml
Comment (paologallinaharbur) [reply .github/workflows/releaseChart.yaml:4]: leaving this open to remember to add it before merging

File: .github/workflows/releaseChart.yaml
Comment (marcsanmi) [inline .github/workflows/releaseChart.yaml:72]: We can use the `CHART_DIRECTORY` env variable here.

File: README.md
Comment (marcsanmi) [inline README.md:140]: Typo in `newrelic-prometheus-agent'` CHANGELOG link.

File: .github/workflows/releaseChart.yaml
Comment (paologallinaharbur) [reply .github/workflows/releaseChart.yaml:72]: true!

**Review by marcsanmi — APPROVED** [review]
LGTM, but maybe I'd wait for some extra approval of someone with full TI context.

**Review by gsanchezgavier — COMMENTED** [review]
LGTM thanks!

**Pull request merged**
